### PR TITLE
[WIP] Implement i18next (cherry-picked from diaspora)

### PR DESCRIPTION
**Pull request by Zauberstuhl**
- [x] remove Diaspora global variables
- [x] clean up translation functions
- [x] developer test

related to #121

### CODE REVIEW

File: .jshintrc [hunk 1]
```diff
@@ -14,6 +14,10 @@
    "multistr": true,
    "devel": true,
    "globals": {
+      "t":true,
+      "i18n":true,
+      "I18next":true,
+      "Diaspora":true,
```
Comment (Zauberstuhl) [inline .jshintrc]: diaspora var

File: jsxc.lib.js [hunk 2]
```diff
@@ -227,9 +227,15 @@ var jsxc;
             lang = jsxc.options.defaultLang;
          }
 
-         // set language
-         jsxc.l = jsxc.l10n.en;
-         $.extend(jsxc.l, jsxc.l10n[lang]);
+         // initialize i18n translator
+         $.i18n.init({
+           lng: Diaspora.I18n.language,
```
Comment (Zauberstuhl) [inline jsxc.lib.js]: diaspora var. Use jsxc language config!

File: jsxc.lib.js [hunk 3]
```diff
@@ -710,12 +716,7 @@ var jsxc;
       translate: function(text) {
          return text.replace(/%%([a-zA-Z0-9_-}{ .!,?/'@]+)%%/g, function(s, key) {
             var k = key.replace(/ /gi, '_').replace(/[.!,?/'@]/g, '');
-
-            if (!jsxc.l[k]) {
-               jsxc.warn('No translation for: ' + k);
```
Comment (Zauberstuhl) [inline jsxc.lib.js]: do we need the warning? We still need the function for translating the templates.

Comment (sualko) [inline jsxc.lib.js]: Why do we need this function, if we move completely to i18next?

Comment (Zauberstuhl) [inline jsxc.lib.js]: right we should use the template function from i18next

**Comment by sualko** [comment]
Are you working in this, or should I continue?

**Comment by Zauberstuhl** [comment]
Yes, it is on my ToDo-list. On the weekend I am on FOSDEM in Brussels maybe I get some time to write it up but if it is urgent feel free to continue.

What is the release plan anyway?

**Comment by sualko** [comment]
Originally I thought we can release 1.1-beta with code reorganisation and i18n at the weekend (stable one week after), but maybe this huge changes are better reflected in a major version upgrade. So I will put both in the 2.0 milestone, which will be hopefully released at the end of the examination period around end of February.

**Comment by Zauberstuhl** [comment]
I have more then enough time in February to finish this PR :)

**Comment by Zauberstuhl** [comment]
Rebased. Awaiting review

**Comment by sualko** [comment]
I'm busy until Monday, so I will review it next week.

File: jsxc.lib.js [hunk 4]
```diff
@@ -3045,55 +3035,53 @@ var jsxc;
...
 
          if (typeof (ret) === 'string') {
-            ret = jsxc.translate(ret);
-
             ret = ret.replace(/\{\{([a-zA-Z0-9_\-]+)\}\}/g, function(s, key) {
                return (typeof ph[key] === 'string') ? ph[key] : s;
             });
 
-            return ret;
+            return $(ret).i18n().text();
```
Comment (sualko) [inline jsxc.lib.js]: I had to replace this line with <code>return $('<div>').append($(ret).i18n()).html();</code>.

File: jsxc.lib.js [hunk 5]
```diff
@@ -223,9 +223,15 @@ var jsxc;
...
-         // set language
-         jsxc.l = jsxc.l10n.en;
-         $.extend(jsxc.l, jsxc.l10n[lang]);
+         // initialize i18n translator
+         $.i18n.init({
+           lng: lang,
+           fallbackLng: 'en',
+           resStore: I18next,
+           // use localStorage and set expiration to a day
+           useLocalStorage: true,
```
Comment (sualko) [inline jsxc.lib.js]: If we use resStore, we don't have to use the local storage, or not? Why are we not using the ajax loading?

**Comment by sualko** [comment]
What do you think about [momentjs](http://momentjs.com) for message timestamps?

Why did you include all the empty/null translation files?

- [ ] convert quotes to apostrophes

**Comment by Zauberstuhl** [comment]
Thanks a lot for fixing that! I am currently very busy :(

**Pull request closed without merging**